apps/rpc/src/modules/article/article-service.ts

```diff
@@ -137,50 +169,63 @@ export function getArticleService(
         .map(({ article }) => article)
         .slice(0, 10)
     },
-    async getFeatured(handle) {
-      const articles = await articleRepository.getFeatured(handle)
-      return articles.sort((a, b) => compareDesc(a.updatedAt, b.updatedAt))
+
+    async findFeatured(handle) {
+      const articles = await articleRepository.findFeatured(handle)
+      return articles.toSorted((a, b) => compareDesc(a.updatedAt, b.updatedAt))
     },
+
     async getTags(handle) {
-      const tags = await articleTagRepository.getAll(handle)
-      return tags.sort((a, b) => compareAsc(a.name, b.name))
+      const tags = await articleTagRepository.findMany(handle)
+      return tags.toSorted((a, b) => compareAsc(a.name, b.name))
     },
+
     async addTag(handle, articleId, tag) {
-      const match = await articleRepository.getById(handle, articleId)
-      if (!match) {
-        throw new NotFoundError(`Article(ID=${articleId}) not found`)
-      }
-      let name = await articleTagRepository.getByName(handle, tag)
+      await requireArticleExists(handle, articleId)
+
+      let name = await articleTagRepository.findByName(handle, tag)
       if (name === null) {
         name = await articleTagRepository.create(handle, tag)
       }
+
       return await articleTagLinkRepository.add(handle, articleId, name.name)
     },
+
     async removeTag(handle, articleId, tag) {
-      const match = await articleRepository.getById(handle, articleId)
-      if (!match) {
-        throw new NotFoundError(`Article(ID=${articleId}) not found`)
-      }
+      await requireArticleExists(handle, articleId)
+
       await articleTagLinkRepository.remove(handle, articleId, tag)
-      const articlesWithTag = await articleRepository.getByTags(handle, [tag])
+
+      const articlesWithTag = await articleRepository.findManyByTags(handle, [tag])
       const isTagStillInUse = articlesWithTag.length !== 0
+
       if (!isTagStillInUse) {
         await articleTagRepository.delete(handle, tag)
       }
     },
+
     async setTags(handle, articleId, tags) {
-      const currentTags = (await articleTagRepository.getAllByArticle(handle, articleId)).map((tag) => tag.name)
-      const tagsToAdd = tags.filter((tag) => !currentTags.includes(tag))
-      const tagsToRemove = currentTags.filter((tag) => !tags.includes(tag))
-      const removePromises = tagsToRemove.map(async (tag) => this.removeTag(handle, articleId, tag))
+      await requireArticleExists(handle, articleId)
```

Comment:
This piece of code does nothing, and does not ensure the article actually exists